### PR TITLE
JBPM-9840 unable to stop KIE container via BC UI with error Not suppo…

### PR DESCRIPTION
**Pull request by bxf12315**
…rted for Web Socket implementation

**Thank you for submitting this pull request**

**JIRA**: _(please edit the JIRA link if it exists)_ 

[https://issues.redhat.com/browse/JBPM-9840](JBPM-9840)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* paste the link(s) from GitHub here
* link 2
* link 3 etc.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins run fdb

File: kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebSocketKieServerClient.java [hunk 1]
```diff
@@ -92,6 +93,14 @@ public WebSocketKieServerClient(String url) {
...
     }
 
+    private List<?> safeList(List<?> list) {
+        if (list == null) {
+
+            return new ArrayList<Object>();
+        }
+        return new ArrayList<Object>(list);
+    }
+
```
Comment (nmirasch) [inline kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebSocketKieServerClient.java:103]: Maybe replace this function by something like this  Optional.ofNullable(list).orElse(Collections.emptyList());

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by nmirasch** [comment]
@bxf12315 I see the behaviour is like QueryServiceClientImpl, and I guess that is the expected. 
I'll try that when we got the green but  would you please add some test?

File: kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebSocketKieServerClient.java (hunk 1 above)
Comment (bxf12315) [reply kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebSocketKieServerClient.java:103]: ok

**Comment by nmirasch** [comment]
@cristianonicolai  I'm not really familiar with the Web Socked implementation. Would you mind to take a look on this one?  to see if this fix about sending the query to all sessions, makes sense. Thanks!

File: kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebSocketKieServerClient.java (hunk 1 above)
Comment (cristianonicolai) [reply kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebSocketKieServerClient.java:103]:
```suggestion
return list == null ? emptyList() : list
```

File: kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebSocketKieServerClient.java [hunk 2]
```diff
@@ -277,15 +286,30 @@ public QueryDefinition getQuery(String queryName) {
...
                     return throwUnsupportedException();
                 }
-                
+
                 @Override
                 public List<ProcessInstance> findProcessInstancesByContainerId(String containerId, List<Integer> status, Integer page, Integer pageSize, String sort, boolean sortOrder) {
-                    return throwUnsupportedException();
+                    ProcessInstanceList result = null;
+                    CommandScript script = new CommandScript(Collections.singletonList((KieServerCommand) new DescriptorCommand("QueryService", "getProcessInstancesByDeploymentId", new Object[]{containerId, safeList(status), page, pageSize, sort, sortOrder})));
+                    ServiceResponse<ProcessInstanceList> listResponse = (ServiceResponse<ProcessInstanceList>) sendCommandToAllSessions(script, new WebSocketServiceResponse(true, (message) -> {
```
Comment (cristianonicolai) [inline kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebSocketKieServerClient.java:294]: use `sendCommand`instead as this is a query and not a mutating call.

File: kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebSocketKieServerClient.java [hunk 3]
```diff
@@ -277,15 +286,30 @@ public QueryDefinition getQuery(String queryName) {
                 public List<ProcessInstance> findProcessInstancesByCorrelationKey(CorrelationKey correlationKey, Integer page, Integer pageSize) {
                     return throwUnsupportedException();
                 }
-                
+
                 @Override
                 public List<ProcessInstance> findProcessInstancesByContainerId(String containerId, List<Integer> status, Integer page, Integer pageSize, String sort, boolean sortOrder) {
-                    return throwUnsupportedException();
+                    ProcessInstanceList result = null;
```
Comment (cristianonicolai) [inline kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebSocketKieServerClient.java:292]: result is never assigned

**Review by cristianonicolai — COMMENTED** [review]
@bxf12315 , please also add tests

File: kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebSocketKieServerClient.java (hunk 2 above)
Comment (bxf12315) [reply kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebSocketKieServerClient.java]: Done

File: kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebSocketKieServerClient.java (hunk 3 above)
Comment (bxf12315) [reply kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/client/WebSocketKieServerClient.java]: Done

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by bxf12315** [comment]
Jenkins retest this

**Review by sutaakar — APPROVED** [review]
Looks good to me.

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by cristianonicolai** [comment]
Jenkins run fdb

**Comment by cristianonicolai** [comment]
@bxf12315 can you add tests please

**Review by cristianonicolai — CHANGES_REQUESTED** [review]
would consider a different approach as the one applied on https://github.com/kiegroup/kie-wb-common/pull/3671. This is a side effect of another issue.

**Comment by mareknovotny** [comment]
is this still valid PR or we can close it?  @bxf12315 @cristianonicolai

**Comment by bxf12315** [comment]
I think it's can be close.
Thanks

**Pull request closed without merging**